data/trie/interceptedNode.go

```diff
@@ -79,3 +79,9 @@ func (inTn *InterceptedTrieNode) IsInterfaceNil() bool {
 func (inTn *InterceptedTrieNode) EncodedNode() []byte {
 	return inTn.encNode
 }
+
+// CreateEndOfProcessingTriggerNode changes the hash of the current node by appending the hash to the current hash.
+// This construction will be used to trigger the end of processing for all of the received data
+func (inTn *InterceptedTrieNode) CreateEndOfProcessingTriggerNode() {
+	inTn.hash = append(inTn.hash, inTn.hash...)
```

Comment:
This is a bit messy and racy (calling append concurrently will yield unexpected results). Why not using a chan struct{}? Way safer and cleaner.

Reply:
The current architecture does not allow for a chan. Added a todo to use a channel at the next refactoring, and added a mutex to protect from race inconsistencies.

Reply:
ok